aula/apps/sortides/filters.py

```diff
@@ -118,6 +158,157 @@ class Meta:  # pylint: disable=too-few-public-methods
+    """
+    Filtre per a la llista de sortides o pagaments.
+    """
+
+    professor = django_filters.ModelChoiceFilter(
+        queryset=None,  # Temporarily set to None
+        method="filter_by_professor",
+        label="Professor que proposa",
+    )
+    departament_que_organitza = django_filters.ModelChoiceFilter(
+        queryset=None,  # Temporarily set to None
+    )
+    tipus = django_filters.ChoiceFilter(
+        choices=Sortida.TIPUS_ACTIVITAT_CHOICES, label="Tipus"
+    )
+
+    titol = django_filters.CharFilter(lookup_expr="icontains", label="Títol")
+    ambit = django_filters.CharFilter(lookup_expr="icontains", label="Àmbit")
+
+    termini_pagament_desde = django_filters.DateFilter(
+        field_name="termini_pagament",
+        lookup_expr="gte",
+        label="Termini pagament posterior a",
+        widget=DateTextImput(),
+    )
+
+    termini_pagament_fins = django_filters.DateFilter(
+        field_name="termini_pagament",
+        lookup_expr="lte",
+        label="Termini pagament anterior a",
+        widget=DateTextImput(),
+    )
+
+    # init
+    def __init__(self, *args, **kwargs):
+        """Init: set queryset and other settings"""
+        super(PagamentFilter, self).__init__(*args, **kwargs)
+
+        Professor = apps.get_model("usuaris", "Professor")
+        self.filters["professor"].queryset = (
+            Professor.objects.all()
+        )  # Set queryset dynamically
+
+        Departament = apps.get_model("usuaris", "Departament")
+        self.filters["departament_que_organitza"].queryset = Departament.objects.all()
+
+        # Crispy Forms Helper
+        self.helper = FormHelper()
+        self.helper.form_method = "GET"
+        self.helper.form_class = "form-inline"
+        self.helper.layout = Layout(
+            Fieldset(
+                "Criteris de cerca",
+                Div(
+                    Field("estat", css_class="form-control form-control-sm"),
+                    css_class="col-lg-4 small",
+                ),
+                Div(
+                    Field(
+                        "tipus_de_pagament", css_class="form-control form-control-sm"
+                    ),
+                    css_class="col-lg-4 small",
+                ),
+                Div(
+                    Field("titol", css_class="form-control form-control-sm"),
+                    css_class="col-lg-4 small",
+                ),
+                Div(
+                    Field("ambit", css_class="form-control form-control-sm"),
+                    css_class="col-lg-4 small",
+                ),
+                Div(
+                    Field(
+                        "departament_que_organitza",
+                        css_class="form-control form-control-sm",
+                    ),
+                    css_class="col-lg-4 small",
+                ),
+                Div(
+                    Field(
+                        "tipus_de_pagament", css_class="form-control form-control-sm"
```

Comment:
Es fa servir dos cops.
WARNING:root:A field should only be rendered once: tipus_de_pagament
NoneType: None
![Captura desde 2025-03-16 11-35-57](https://github.com/user-attachments/assets/2389d1cc-321d-4d91-84e3-1e59e702f3f7)

Reply:
Hola, gràcies per la review. Fixat al nou commit.